src/functions/edi/inbound/handler.ts

```diff
@@ -102,34 +102,50 @@ export const handler = async (event: any): Promise<Record<string, any>> => {
             receivingPartnerId,
           });
 
-          for (const functionalGroup of interchange.functionalGroups) {
-            const guideIdsForPartnership = transactionSetConfigs.map(
-              (config) => config.guideId
-            );
+          const guideIdsForPartnership = transactionSetConfigs.map(
```

Comment:
moved this outside of the functional groups loop -- no need to do it more than once per interchange